### PR TITLE
[Spike] Snackbar proof of concept

### PR DESCRIPTION
**Pull request by TimRoe**
## Description of Change

Created a proof of concept for the Snackbar component that works in both Storybook and flagship. This ticket mostly lays some groundwork, significant refinement will be made on implementation ticket for Snackbar including creating the custom display and getting into specifics where this PR leaves fairly general properties copied from flagship.

#### Testing Packages

- [0.20.4-alpha.0](https://www.npmjs.com/package/@department-of-veterans-affairs/mobile-component-library/v/0.20.4-alpha.0)

### Screenshots/Video

Storybook (with 3 second duration for testing):

https://github.com/user-attachments/assets/55fb1cca-795f-4e3d-a72b-25fd6fadc6a9

Flagship (with 3 second duration for testing):

https://github.com/user-attachments/assets/8f7a30ea-7214-42bc-920b-2e62c8daefcb

## Testing

Validated it worked as expected in iOS in Storybook. Validated it worked as expected in flagship (with minor fixes in node_modules from alpha build that were retroactively made on the branch but no new alpha was published).

Current implementation did not work in Web Storybook. That will be deferred to later tickets. In theory, it should be possible as the react-native-toast-notifications package documentation includes a web demonstration of it working. It may require remedial work specifically for Web Storybook though. Worst case scenario, we can just drop the custom Snackbar UI in stories and not have it "pop up" since Storybook is for documentation purposes and that is likely sufficient.

- [x] Tested on iOS 
- [x] Tested on Android 
- [x] Tested on Web
   - Checking off, but it did not work in Web

## PR Checklist
Code reviewer validation:
- General
	- [x] PR is linked to ticket(s)
	- [x] PR has `changelog` label applied if it's to be included in the changelog
	- [x] Acceptance criteria: 
		- All satisfied _or_
		- Documented reason for not being performed _or_
		- Split to separate ticket and ticket is linked by relevant AC(s)
	- [x] Above PR sections adequately filled out
	- [x] If any breaking changes, in accordance with the [pre-1.0.0 versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library#versioning-policy): a CU ticket has been created for the VA Mobile App detailing necessary adjustments with the package version that will be published by this ticket
- Code
	- [x] Tests are included if appropriate (or split to separate ticket)
	- [x] New functions have proper TSDoc annotations

## Publish

While changes do warrant a new version [per the versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library#versioning-policy), it is postponed from this ticket as it's just a proof of concept laying the groundwork.

### CODE REVIEW

File: packages/components/global.d.ts [hunk 1]
```diff
@@ -8,3 +8,7 @@ declare module '*.svg' {
   const content: React.FC<SvgProps>
   export default content
 }
+
+type SnackbarType = import('./src/components/Snackbar/Snackbar').SnackbarType
+// eslint-disable-next-line no-var
+declare var snackbar: SnackbarType
```
Comment (narin) [inline packages/components/global.d.ts:14]: Does `const` not work here?

**Review by narin — APPROVED** [review]
Looking like a good start to this.

File: packages/components/global.d.ts (hunk 1 above)
Comment (TimRoe) [reply packages/components/global.d.ts:14]: That is correct per [StackOverflow answer here](https://stackoverflow.com/a/56984941/13261302) and corresponding [TS documentation for global variables](https://devblogs.microsoft.com/typescript/announcing-typescript-3-4/#type-checking-for-globalthis):
> Note that global variables declared with `let` and `const` don’t show up on `globalThis`.

Admittedly didn't test if that had changed more recently.

Comment (TimRoe) [reply packages/components/global.d.ts:14]: As a circle-back on this. I tried using `const` on my branch implementing it and it does work to launch the snackbar, but gives a TS error on `globalThis.snackbar` so seems the StackOverflow answer is correct that globalThis only recognizes `var`.

**Pull request merged**